Detectors/TRD/reconstruction/src/DataReaderTask.cxx

```diff
@@ -61,63 +73,77 @@ void DataReaderTask::run(ProcessingContext& pc)
   auto device = pc.services().get<o2::framework::RawDeviceService>().device();
   auto outputRoutes = pc.services().get<o2::framework::RawDeviceService>().spec().outputs;
   auto fairMQChannel = outputRoutes.at(0).channel;
-  int inputcount = 0;
+  //auto datadesc = (mUserDataDescription == o2::header::gDataDescriptionInvalid) ? o2::header::gDataDescriptionRawData : mUserDataDescription;
+  auto datadesc = (mDataDesc == o2::header::gDataDescriptionInvalid.str) ? o2::header::gDataDescriptionRawData : mUserDataDescription;
```

Comment:
I don't see where do you set the `mUserDataDescription` (except initialization by gDataDescriptionInvalid) but what do you need it at all? Don't you have all needed info in the `mDataDesc` ? And I don't see how do you use at all this datadesc? You still hardcode the `RAWDATA`.
Could you explain in which cases the input will *not* be the `RAWDATA`, what it is will be and who will handle the `RAWDATA` in this case (I guess the StfBuilder always sends the `RAWDATA`?) ?

Concerning the treatment of `trd-datareader-inputspec` option: in the DataReader you are treating it is a global workflow option, in the DataReaderTask.cxx: as a local device option. This may pose a problem since in general these are different type of options. Please don't mix them, since it is in reality a workflow option (defined in DataReader.cxx) and you are parsing it there, you better simply pass its parsed value to DataReaderTask, as you do for other arguments in 
https://github.com/AliceO2Group/AliceO2/blob/3eb01802d28a3667adb1155b2825c4f9c8fd95d1/Detectors/TRD/reconstruction/src/DataReader.cxx#L74

Finally, we are placing the workflows and DPL divices classes in separate workflow directory, could eventually (in different PR) move DataReader and DataReaderTask to such dir.?

Reply:
Original idea was to have RAWDATA and CRAWDATA for without and with the compressor. Will clean up now and only do RAWDATA, the parameter list kept failing for a string or char* or various other things i tried, so simply reverted to putting the option code you see in the init. Will simply change it all to RAWDATA and remove the inputspec option.

Reply:
OK, because if you would expect the CRAWDATA, the protection would need to be moved to the processor which converts RAWDATA to CRAWDATA. The point is to have a protection at the 1st instance where the TF might be missing.

Reply:
cool that will go in another pr